src/AlexaVoxCraft.MediatR.Generators/Generators/SymbolDiscovery.cs

```diff
@@ -0,0 +1,200 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using AlexaVoxCraft.MediatR.Generators.Models;
+using Microsoft.CodeAnalysis;
+
+namespace AlexaVoxCraft.MediatR.Generators.Generators;
+
+internal static class SymbolDiscovery
+{
+    private const string AlexaHandlerAttributeName = "AlexaVoxCraft.MediatR.Annotations.AlexaHandlerAttribute";
+    private const string IRequestHandlerName = "AlexaVoxCraft.MediatR.IRequestHandler";
+    private const string IDefaultRequestHandlerName = "AlexaVoxCraft.MediatR.IDefaultRequestHandler";
+    private const string IPipelineBehaviorName = "AlexaVoxCraft.MediatR.Pipeline.IPipelineBehavior";
+    private const string IExceptionHandlerName = "AlexaVoxCraft.MediatR.Pipeline.IExceptionHandler";
+    private const string IRequestInterceptorName = "AlexaVoxCraft.MediatR.Pipeline.IRequestInterceptor";
+    private const string IResponseInterceptorName = "AlexaVoxCraft.MediatR.Pipeline.IResponseInterceptor";
+    private const string IPersistenceAdapterName = "AlexaVoxCraft.MediatR.Attributes.Persistence.IPersistenceAdapter";
+
+    public static (RegistrationModel model, ImmutableArray<DiagnosticInfo> diagnostics) BuildModel(ImmutableArray<INamedTypeSymbol> symbols)
```

Comment:
Might want to use EquatableArray here:
```suggestion
    public static (RegistrationModel model, EquatableArray<DiagnosticInfo> diagnostics) BuildModel(ImmutableArray<INamedTypeSymbol> symbols)
```

Reply:
But that would only be needed if you call this before `RegisterSourceOutput` (which you should also do 😄).